src/pkg/egress/syslog/https_batch.go

```diff
@@ -183,16 +239,90 @@ func (w *HTTPSBatchWriter) startSender() {
 	}
 }
 
+// Close gracefully shuts down the HTTPSBatchWriter and stops all workers.
 func (w *HTTPSBatchWriter) Close() error {
-	close(w.quit)
-	w.wg.Wait() // Ensure sender finishes processing before closing
-	close(w.msgChan)
+	w.closeOnce.Do(func() {
+		// Close msgChan first to stop Write() from sending more messages
+		close(w.msgChan)
+
+		// Signal startSender to quit after draining remaining messages
+		close(w.quit)
+
+		// Wait for startSender to finish processing
+		w.wg.Wait()
+
+		if w.dispatcher != nil {
+			w.dispatcher.stop()
+		}
+	})
 	return nil
 }
 
+// httpBatchClient creates a fasthttp.Client with custom timeout configurations.
 func httpBatchClient(netConf NetworkTimeoutConfig, tlsConf *tls.Config) *fasthttp.Client {
 	client := httpClient(netConf, tlsConf)
 	client.MaxIdleConnDuration = 30 * time.Second
 	client.MaxConnDuration = 30 * time.Second
 	return client
 }
+
+// batch represents a batch of bytes to send with message count.
+type batch struct {
+	data     []byte
+	msgCount float64
+}
+
+// worker receives batches from input channel and sends them with retries.
+type worker struct {
+	id      int
+	input   chan batch
+	retryer *Retryer
+	writer  *HTTPSBatchWriter
+}
+
+// start listens on input channel and sends batches with retry logic.
+func (w *worker) start() {
+	for b := range w.input {
+		ctx := context.Background()
+		if w.retryer.binding.Context != nil {
+			ctx = w.retryer.binding.Context
+		}
+		w.retryer.Retry(ctx, b.data, b.msgCount, w.writer.sendHttpRequest)
+	}
+}
+
+// dispatcher distributes batches to workers in a round-robin fashion.
+type dispatcher struct {
+	workers    []*worker
+	nextWorker int
+	mu         sync.Mutex
+}
+
+// dispatch sends a batch to the next worker in round-robin order.
+func (d *dispatcher) dispatch(b batch) {
```

Comment:
this actually round robins workers, and does not lock a thread if it retries, working more like a general choke on the throughput once one syslog drain retries.
This is the opposite of what we need.